frontend/app/.server/domain/services/api-client.ts

```diff
@@ -83,5 +121,24 @@ export const apiClient = {
     }
   },
 
-  // TODO: add post, put, delete methods here in the future following the same pattern.
+  async delete<T>(path: string, context: string, token?: string, model?: T): Promise<Result<T, AppError>> {
```

Comment:
Request and response data type can be different.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Methods/DELETE

```suggestion
  async delete<TRequestData, TResponseData = unknown>(path: string, context: string, token?: string, data?: TRequestData): Promise<Result<TResponseData, AppError>> {
```